pghoard/rohmu/compat.py

```diff
@@ -1,59 +1,14 @@
-# rohmu - compatibility functions to support older Python 3 versions
+# rohmu - This module used to contain compatible implementations for older
+# python version.
 #
-# Code mostly copied from Python 3.4.3 under the Python license.
+# Since we don't support those older versions anymore, this module could be removed,
+# but as rohmu is used outside pghoard itself better keep the imports and throw
+# a deprecation warning.
 
-# suppress: call or simulate Pyhon 3.4 contextlib.suppress
+from contextlib import suppress  # pylint: disable=unused-import
+from os import makedirs  # pylint: disable=unused-import
 
-try:
-    from contextlib import \
-        suppress  # pylint: disable=import-error, no-name-in-module, unused-import
-except ImportError:
-
-    class suppress:
-        """Context manager to suppress specified exceptions
-
-        After the exception is suppressed, execution proceeds with the next
-        statement following the with statement.
-
-             with suppress(FileNotFoundError):
-                 os.remove(somefile)
-             # Execution still resumes here if the file was already removed
-        """
-        def __init__(self, *exceptions):
-            self._exceptions = exceptions
-
-        def __enter__(self):
-            pass
-
-        def __exit__(self, exctype, excinst, exctb):
-            # Unlike isinstance and issubclass, CPython exception handling
-            # currently only looks at the concrete type hierarchy (ignoring
-            # the instance and subclass checking hooks). While Guido considers
-            # that a bug rather than a feature, it's a fairly hard one to fix
-            # due to various internal implementation details. suppress provides
-            # the simpler issubclass based semantics, rather than trying to
-            # exactly reproduce the limitations of the CPython interpreter.
-            #
-            # See http://bugs.python.org/issue12029 for more details
-            return exctype is not None and issubclass(exctype, self._exceptions)
-
-
-# makedirs: call or simulate Pyhon 3.4.1 os.makedirs
-# Earlier Python versions raise an error if exist_ok flag is set and the
-# directory exists with different permissions.
-
-import os
-import sys
-
-if sys.version_info >= (3, 4, 1):
-    makedirs = os.makedirs
-else:
-
-    def makedirs(path, mode=0o777, exist_ok=False):
-        if not exist_ok:
-            return os.makedirs(path, mode)
-        try:
-            return os.makedirs(path, mode)
-        except FileExistsError:
-            pass
-        return None
+raise DeprecationWarning(
+    "pghoard.rohmu.compat is deprecated, you should import "
+    "from the standard library directly instead"
+)
```

Comment:
Raising it will immediately break the calling code, not really a warning.

I've never used the before, but I think you need to use https://docs.python.org/3/library/warnings.html#warnings.warn instead of raising yourself (by default it will only print but users can configure it to raise).

Reply:
You're right, fixed.